app/Views/Templates/components/forms/text-input.blade.php

```diff
@@ -0,0 +1,48 @@
+@props([
+    'inputType' => 'text',
+    'id' => '',
+    'name' => '',
+    'placeholder' => '',
+    'label' => '',
+    'size' => '',
+    'inputState' => '',
+    'value' => '',
+    'leadingVisual' => '',
+    'trailingVisual' => '',
+])
+
+@php
+    $sizeClass = $size ? 'input-'.$size : '';
+    $stateClass = $inputState ? 'input-'.$inputState : '';
+@endphp
+
+<div class='par relative w-full max-w-xs'>
+    @if($label)
+        <label for="{{ $id }}">
+            <span class="label-text">{{ $label }}</span>
+        </label>
+    @endif
+
+    <div class="relative">
+        @if($leadingVisual)
+            <span class="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
+                <i class="{{ $leadingVisual }}"></i>
+            </span>
+        @endif
+
+        <input 
+            type="{{ $inputType }}" 
+            id="{{ $id }}" 
+            name="{{ $name }}" 
+            placeholder="{{ $placeholder }}" 
+            value="{{ $value }}" 
+            {{$attributes->merge(['class' => 'input input-bordered '.$sizeClass.' '.$stateClass.' '.'w-full max-w-xs '.($leadingVisual ? 'pl-10' : '').($trailingVisual ? 'pr-10' : '')])}}
+        />
+
+        @if($trailingVisual)
+            <span class="absolute inset-y-0 right-0 pr-3 flex items-center pointer-events-none">
+                <i class="{{ $trailingVisual }}"></i>
+            </span>
+        @endif
+    </div>
```

Comment:
In additon to the label can you also add the bottom left text and top right text as outlined here: https://daisyui.com/components/input/#with-form-control-and-labels.

Text on the bottom should be an attribute called "caption" top right probebly just label-right. 
The caption text should have a state class as well so that if we use it for validation messages and we send in "danger" as the state it becomes "text-danger"